server/server.js

```diff
@@ -86,5 +147,42 @@ module.exports = function(port, db, githubAuthoriser) {
         });
     });
 
+    app.get("/api/conversations", function (req, res) {
+        users.findOne({
+            _id: req.session.user
+        }, function (err, user) {
+            if (!err) {
+                conversations.find({
+                    userids: user._id
+                }).toArray(function (err, relevantConversations) {
+                    res.json(relevantConversations);
+                });
+            } else {
+                res.sendStatus(500);
+            }
+        });
+
+    });
+
+    app.post("/api/newMessage", function (req, res) {
+        users.findOne({
+            _id: req.session.user
+        }, function (err, user) {
+            if (!err) {
+                var newMessage = new message(user._id, req.body.messageText);
+                conversations.findOneAndUpdate({
+                    _id: req.body.conversationId
+                }, {
+                    $push: {
+                        messages: newMessage
+                    }
+                });
+                res.sendStatus(201);
+            } else {
+                res.sendStatus(500);
```

Comment:
Might want to send some sort of explanation here. 500 is usually used as a catch-all generic error. In this case, it's more of a 404: User not found.